File: benchmark0/cpp/int_overflow.cpp
// C++ silently promotes uint8_t operands to int before addition.
// No overflow occurs here — the result always fits in int.
// A naive Rust translation using `u8 + u8` would panic on overflow in debug mode.
#include <cstdint>

int add_bytes(uint8_t a, uint8_t b) {
    return a + b;
}

File: benchmark0/rust/int_overflow.rs
// Cast to i32 before adding, making the integer promotion explicit.
// Unlike C++, Rust does not silently widen u8 to a larger integer type.
// The result is always in [0, 510] so i32 is sufficient and no panic is possible.

pub fn add_bytes(a: u8, b: u8) -> i32 {
    (a as i32) + (b as i32)
}
